src/solvers/file_solution_history.C

```diff
@@ -164,19 +165,27 @@ void FileSolutionHistory::retrieve(bool is_adjoint_solve, Real time)
   if(is_adjoint_solve)
   {
     // Reading in the primal xdas overwrites the adjoint solution with zero
-    // So swap to retain the old adjoint solution
+    // So swap to retain the adjoint and old adjoint vectors
     for (auto j : make_range(_system.n_qois()))
     {
       dual_solution_copies[j] = _system.get_adjoint_solution(j).clone();
+
+      std::string old_adjoint_solution_name = "_old_adjoint_solution";
```

Comment:
Weird, when I reload the github page I'm not seeing the comment that this comment is referencing.  For the record my other complaint was about History classes assuming the number of adjoint solution vectors that TimeSolver classes need.  When we go higher order we'll be adding more vectors, so we'll need some interface to let the TimeSolver classes communicate the complete list, we can't just hard-code it.

This is a big enough advance that it doesn't need to be fixed in this PR, we just need to start thinking about it before we dig ourselves into some hole where it can't eventually be fixed.